api/nodes/serializers.py

```diff
@@ -1456,7 +1456,11 @@ class Meta:
 
     @staticmethod
     def get_id(obj):
-        return f'{obj.node._id}:{obj.provider}'
+        from addons.base.models import BaseAddonSettings
+        if issubclass(type(obj.provider_settings), BaseAddonSettings):
+            return f'{obj.node._id}:{obj.provider}'
+        else:
+            return obj.provider_settings.gv_data.resource_id
```

Comment:
For `NodeStorageProvider` endpoint, then `id` of files providers for addons will be the `id` of the `ConfiguredStorageAddon` as serialized from GV.